### PR TITLE
[gettext-libintl] Changes to allow compiling osx with osxcross

### PR DESCRIPTION
**Pull request by nonameentername**
- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/gettext-libintl/portfile.cmake [hunk 1]
```diff
@@ -77,6 +77,10 @@ if(VCPKG_TARGET_IS_WINDOWS)
     endif()
 endif()
 
+if(VCPKG_TARGET_IS_OSX AND VCPKG_CROSSCOMPILING)
+    list(APPEND OPTIONS --host ${TARGET_TRIPLET}-darwin)
+endif()
```
Comment (dg0yt) [inline ports/gettext-libintl/portfile.cmake:82]: I think this isn't code for the specific port but for `scripts/vcpkg_configure_make.cmake`. (Which is more expensive to change, affecting all triplets and many ports.)

Comment (MonicaLiu0311) [reply ports/gettext-libintl/portfile.cmake:82]: Ping @nonameentername

**Comment by dg0yt** [comment]
@MonicaLiu0311 IMO you shouldn't update the branch as long as the PR is understood as taking the wrong road. Rebuilding `gettext-libintl` is expensive. And this rebuild hit CI in a high-load time.

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 IMO you shouldn't update the branch as long as the PR is understood as taking the wrong road. Rebuilding `gettext-libintl` is expensive. And this rebuild hit CI in a high-load time.

The remaining triplets have been cancelled, and this PR will be closed if there is no response next week.

**Pull request closed without merging**